pkg/cloudprovider/providers/aws/aws_test.go

```diff
@@ -1073,6 +1074,33 @@ func TestFindInstancesByNodeNameCached(t *testing.T) {
 	}
 }
 
+func TestGetInstanceByNodeNameBatching(t *testing.T) {
+	awsServices := NewFakeAWSServices()
+	c, err := newAWSCloud(strings.NewReader("[global]"), awsServices)
+	assert.Nil(t, err, "Error building aws cloud: %v", err)
+	var tag ec2.Tag
+	tag.Key = aws.String(TagNameKubernetesClusterPrefix + TestClusterId)
+	tag.Value = aws.String("")
+	tags := []*ec2.Tag{&tag}
+	nodeNames := []string{}
+	for i := 0; i < 200; i++ {
```

Comment:
Yeah "some", :-) . Because we still don't test if partitioning correctly splits the results too.   I can extract this into a separate function easily enough. Will open a github issue for myself.